src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -183,6 +213,10 @@ impl<'tcx> NewPermission {
                 // Weak protector for boxes
                 ProtectorKind::WeakProtector
             }),
+            do_a_write: start_mut_ref_on_fn_entry_as_active
+                && is_protected
+                && !current_function_involves_raw_pointers
+                && matches!(ref_mutability, Some(Mutability::Mut)),
```

Comment:
Instead of adding a new field, this should set both `freeze_perm` and `nonfreeze_perm` to `Active`.

That should then be used as a signal to control the `AccessKind` for the (already existing) initial access logic in `tb_reborrow`.